Updating smoke test, as per the composer functionality updated for ignore minor/patch updates

thavaahariharangit opened the pull request:
This changes will fix the https://github.com/dependabot/dependabot-core/actions/runs/9472992031/job/26099449614?pr=9974 issue.

Due to minor patch filter issue, we have updated dependabot core functionality. That broke the corresponding smoke test, This PR contains the fix for that.

jurre reviewed (commented):
I am confused by all of this, it seems like this test expected actions to be updated? That seems like it's not what the test was supposed to be testing, given its for the composer ecosystem, and, why do the changes all of a sudden cause this to fail?

thavaahariharangit commented:
> I am confused by all of this, it seems like this test expected actions to be updated? That seems like it's not what the test was supposed to be testing, given its for the composer ecosystem, and, why do the changes all of a sudden cause this to fail?

After seeing Garry's PR failing on Composer Smoke https://github.com/dependabot/dependabot-core/actions/runs/9472992031/job/26099449614?pr=9974 , I was checking smoketest repo workflow by creating this PR. In the work flow I saw the difference between smoke-composer.yml (expected result file that I have updated yesterday) and the actual results.yml(actual result generated by work flow) having the difference. So that's why created this PR in-order fix that difference. 

According to the new logic that we have implemented. every dependency will try to upgrade to the lates allowable in the registry. If that is not compatible. there will not be any PR generated.

jakecoffman commented:
Closing, see #209 and #210 for background.
